app/controllers/time_slot_controller.rb

```diff
@@ -40,13 +40,8 @@ def schedule
   end
 
   def cancel
-    Appointment.transaction do
-      @current_patient.appointments
-        .where(id: cancel_params[:appointment_id])
-        .update_all(patient_id: nil)
-
-      @current_patient.update(last_appointment: nil)
-    end
+    @appointment = @current_patient.appointments.find_by!(id: cancel_params[:appointment_id])
+    @appointment.update!(patient_id: nil)
```

Comment:
This will make a `SELECT` and an `UPDATE`. We can have a single `UPDATE` with the previous code:

```suggestion
    @current_patient.appointments
      .where(id: cancel_params[:appointment_id])
      .update_all(patient_id: nil)
```

Reply:
Ah, you're right, nice catch!